File: Test_DataParser.hs
module Test_DataParser where

import DataParser
import Test.HUnit
import ExerciseManager

test_parseStandardDate_returns_a_date_object_that_makes_sense =
  TestCase (assertEqual "Making sure that parseStandardDate works" (Date 4 6 1992) (parseStandardDate "04/06/1992"))

tests = TestList [test_parseStandardDate_returns_a_date_object_that_makes_sense]

main = runTestTT tests

File: test_data_parser.rb
#!/usr/bin/env ruby

require 'test/unit'
require './data_parser'

class TestDataParser < Test::Unit::TestCase
  def test_parse_date_parses_a_standard_date_correctly
    answer = DataParser.date("05/05/13")
    expected = "Date 5 5 2013"

    assert_equal expected, answer
  end
end
